### PR TITLE
[DROOLS-3162] Implement full page width support for core-grids

### PR DESCRIPTION
**Pull request by danielezonca**
This PR adds the support to have columns with auto width to fit entirely the visible area. All `auto` columns try to fit the entire page except if this means to be shorted than the defined minimum, in this case the grid will be larger than then the visible area 

**How it works:**
- All the columns are `fixed` by default (to be retro-compatible)
- In scesim all `GIVEN` and `EXPECT` columns have `auto` width

@Rikkola @manstis @kkufova @gitgabrio 

PR list:
https://github.com/kiegroup/appformer/pull/543
https://github.com/kiegroup/drools-wb/pull/1003
https://github.com/kiegroup/kie-wb-common/pull/2299

### CODE REVIEW

**Comment by manstis** [comment]
@danielezonca Tests failures are genuinely introduced by this PR...

**Comment by danielezonca** [comment]
@manstis yes I'm looking at them right now :)

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/model/DelegatingGridDataTest.java [hunk 1]
```diff
@@ -350,4 +351,39 @@ public void testDelegateGetRow() {
...
+        uiModel.refreshWidth(0);
+
+        verify(delegate).refreshWidth(eq(0));
+    }
+
+    @Test
+    public void testSetVisibleSizeAndRefresh() {
+        uiModel.setVisibleSizeAndRefresh(0, 0);
+
+        verify(delegate).setVisibleSizeAndRefresh(eq(0), eq(0));
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/model/DelegatingGridDataTest.java]: IMO it's better to use different values, e.g. `0` and `1` to be sure, for example, that the "delegated" method is not simply using the first parameter twice...

Comment (danielezonca) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/model/DelegatingGridDataTest.java]: Done

**Comment by danielezonca** [comment]
@manstis 
I fixed the code so now the tests should run (added an `if` to keep exactly the same behavior for existing columns that have the default `FIXED` width mode).
Btw I saw in at least two classes a similar code
https://github.com/kiegroup/kie-wb-common/blob/11e04f1256ca289988d04cff871e5da2e1acce68/kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/expressions/types/invocation/InvocationGridTest.java#L334-L345
and I think it should be better as general rule to define the behavior of the mock before use it. Wdyt?

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/model/DelegatingGridDataTest.java [hunk 2]
```diff
@@ -350,4 +351,39 @@ public void testDelegateGetRow() {
...
+
+    @Test
+    public void testRefreshWidth() {
+        uiModel.refreshWidth();
+
+        verify(delegate, times(1)).refreshWidth();
+    }
+
+    @Test
+    public void testRefreshWidth1() {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/model/DelegatingGridDataTest.java]: Sorry @danielezonca would you mind making this test name more descriptive, like IDK `testRefreshWidthWithSize()`?

Comment (danielezonca) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/model/DelegatingGridDataTest.java]: Sure, I agree it is better to rename it. Done :)

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/model/DelegatingGridData.java [hunk 3]
```diff
@@ -259,4 +259,29 @@ public void appendRow(final GridRow row) {
...
         return delegate.getRow(rowIndex);
     }
+
+    @Override
+    public boolean refreshWidth() {
+        return delegate.refreshWidth();
+    }
+
+    @Override
+    public boolean refreshWidth(double currentWidth) {
```
[the remark this answers is not in the document]
Comment (danielezonca) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/model/DelegatingGridData.java]: Done

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/model/DelegatingGridData.java [hunk 4]
```diff
@@ -259,4 +259,29 @@ public void appendRow(final GridRow row) {
...
+        return delegate.refreshWidth(currentWidth);
+    }
+
+    @Override
+    public boolean setVisibleSizeAndRefresh(int width, int height) {
```
[the remark this answers is not in the document]
Comment (danielezonca) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/model/DelegatingGridData.java]: Done

**Comment by danielezonca** [comment]
@manstis 
Can you please squash and merge?

**Pull request merged**